engine/src/mac-window.mm

```diff
@@ -2244,7 +2244,7 @@ void MCMacPlatformHandleDrawSync(NSWindow *window)
 		// we enter the runloop to trigger a redraw. This will cause drawRect to be invoked on our view
 		// which in turn will result in a redraw window callback being sent.
 		// The timeout value of 0.02ms is specified to avoid hitting the 60hz redraw limit.
-		if (!s_inside_focus_event && !s_showing_sheet && ![m_delegate inUserReshape])
+		if (int(MCwaitdepth) == 0 && !s_inside_focus_event && !s_showing_sheet && ![m_delegate inUserReshape])
```

Comment:
So what if the wait depth > 1 due to a scripted wait...
e.g.
```
on myRunLoop
    repeat forever
       wait for messages
       ... do stuff ...
    end repeat
end myRunLoop
```
Won't this cause significantly different behavior in event handlers being triggered from the nested wait from when they are triggered outside of it?

Reply:
OK I've revised this to be more specific to `[NSApp sendEvent:...]`. Also switched to using MCPlatformBreakWait as without that the lockout still occurs

Reply:
Isn't that the same though? As sendEvent causes the event to be percolated... i.e.
```
sendEvent
  ... invokes mouse* of view ...
     ... causes LC event handler to run ...
```

Reply:
This has been revised to avoid this test now :)